src/NUnitFramework/framework/Schemas/TestResult.xsd

```diff
@@ -0,0 +1,232 @@
+                    <xs:complexType>
+                      <xs:all>
+                        <xs:element name="filePath" type="xs:string" />
+                        <xs:element name="description" minOccurs="0" type="xs:string" />
+                      </xs:all>
+                    </xs:complexType>
+                  </xs:element>
+                </xs:sequence>
+              </xs:complexType>
+            </xs:element>
+          </xs:sequence>
+        </xs:complexType>
+      </xs:element>
+    </xs:sequence>
+  </xs:group>
+  <xs:attributeGroup name="TestResultBaseAttributeGroup">
+    <xs:attribute name="result" use="required">
+      <xs:simpleType>
+        <!-- NUnit.Framework.Interfaces.TestStatus -->
+        <xs:restriction base="xs:string">
+          <xs:enumeration value="Inconclusive" />
+          <xs:enumeration value="Skipped" />
+          <xs:enumeration value="Passed" />
+          <xs:enumeration value="Warning" />
+          <xs:enumeration value="Failed" />
+        </xs:restriction>
+      </xs:simpleType>
+    </xs:attribute>
+    <xs:attribute name="label" type="xs:string" />
+    <xs:attribute name="site">
+      <xs:simpleType>
+        <!-- NUnit.Framework.Interfaces.FailureSite -->
+        <xs:restriction base="xs:string">
+          <xs:enumeration value="Test" />
+          <xs:enumeration value="SetUp" />
+          <xs:enumeration value="TearDown" />
+          <xs:enumeration value="Parent" />
+          <xs:enumeration value="Child" />
+        </xs:restriction>
+      </xs:simpleType>
+    </xs:attribute>
+    <xs:attribute name="start-time" type="xs:string"/>
+    <xs:attribute name="end-time" type="xs:string" />
+    <xs:attribute name="duration" type="TestDurationType" />
+    <xs:attribute name="asserts" type="NonnegativeInt32" use="required" />
+  </xs:attributeGroup>
+
+  <!-- NUnit.Framework.Api.FrameworkController.RunTests -->
+  <xs:group name="RootResultElementGroup">
+    <xs:sequence>
+      <!-- NUnit.Framework.Api.FrameworkController.InsertEnvironmentElement -->
+      <xs:element name="environment">
+        <xs:complexType>
+          <xs:attribute name="framework-version" type="xs:string" use="required" />
+          <xs:attribute name="clr-version" type="xs:string" use="required" />
+          <xs:attribute name="os-version" type="xs:string" use="required" />
+          <xs:attribute name="platform" type="xs:string" />
+          <xs:attribute name="cwd" type="xs:string" use="required" />
+          <xs:attribute name="machine-name" type="xs:string" />
+          <xs:attribute name="user" type="xs:string" />
+          <xs:attribute name="user-domain" type="xs:string" />
+          <xs:attribute name="culture" type="xs:string" use="required" />
+          <xs:attribute name="uiculture" type="xs:string" use="required" />
+          <xs:attribute name="os-architecture" type="xs:string" use="required" />
+        </xs:complexType>
```

Comment:
I _think_ the xamarin runner uses slightly different fields in its top level test-run properties? Thats possibly the wrong thing to do, but if we’re going to publish this as ‘what will exist’ - I think we should check. (On my phone right now, sorry!)